lib/pacemaker/pcmk_graph_producer.c

```diff
@@ -996,22 +996,24 @@ pcmk__log_transition_summary(const char *filename)
  * \param[in] rsc  Resource whose actions should be added
  */
 void
-native_expand(pe_resource_t *rsc)
+pcmk__add_rsc_actions_to_graph(pe_resource_t *rsc)
 {
-    GList *gIter = NULL;
+    GList *iter = NULL;
 
-    CRM_ASSERT(rsc);
-    pe_rsc_trace(rsc, "Processing actions from %s", rsc->id);
+    CRM_ASSERT(rsc != NULL);
+    pe_rsc_trace(rsc, "Adding actions for %s to graph", rsc->id);
 
-    for (gIter = rsc->actions; gIter != NULL; gIter = gIter->next) {
-        pe_action_t *action = (pe_action_t *) gIter->data;
+    // First add the resource's own actions
+    for (iter = rsc->actions; iter != NULL; iter = iter->next) {
+        pe_action_t *action = (pe_action_t *) iter->data;
 
-        crm_trace("processing action %d for rsc=%s", action->id, rsc->id);
+        crm_trace("Adding action %d for %s to graph", action->id, rsc->id);
         pcmk__add_action_to_graph(action, rsc->cluster);
     }
```

Comment:
If you didn't care about the log message, or you moved the log message into pcmk__add_action_to_graph, this can be:
```
g_list_foreach(rsc->actions, (GFunc) pcmk__add_action_to_graph, rsc->cluster)
```

Reply:
Good point, and this is no longer used elsewhere, so it can be static too. Added commit on top.